ATFConverter/cdli_import.py

```diff
@@ -49,12 +51,155 @@ def __read_file__(file):
+        return output
+
+    @staticmethod
+    def __split_texts__(read_file):
+        """
+        Using read_file, this process separates out tablets based off of
+        metadata in the file.
+        :param read_file: This is the text file that you downloaded from
+        CDLI.
+        :return: List that separates out disparate texts into lists of strings.
+        """
+        output = []
+        for line in read_file:
+            if line.startswith('Primary'):
+                yield output
+                output = []
+            output.append(line)
+        yield output    # issue: prints "[[]" before list
+
+    def texts_within_file(self, read_file):
+        """
+        Using the read_file function, this process utilizes discern_texts
+        and split_texts in order to create a dictionary of texts from the file.
+
+        i.e. either "ARM 01, 001" or "&P254202" for key; the text for value.
+
+        :param read_file: This is the text file that you downloaded from
+        CDLI.
+        :return: Dictionary of separated texts in a file with either the
+        CDLI number or published name of text as key & said text as its value.
+        """
+
+        titles = self.__discern_texts__(read_file)
+        return "\n".join(titles)
+
+    def text_contents(self, read_file):
         """
+        Using the read_file function, this process utilizes discern_texts
+        and split_texts in order to create a dictionary of texts from the file.
+
+        i.e. either "ARM 01, 001" or "&P254202" for key; the text for value.
+
+        :param read_file: This is the text file that you downloaded from
+        CDLI.
+        :return: Dictionary of separated texts in a file with either the
+        CDLI number or published name of text as key & said text as its value.
+        """
+        key = self.__discern_texts__(read_file)
+        value = list(self.__split_texts__(read_file))
+        texts = zip(key, value[1:])
+        text_dict = dict(texts)
+
+        return text_dict
+
+    @staticmethod
+    def text_print(text_contents, key):
+        """
+        Using text_contents, select the respective key you wish to print for
+        tokenization or pretty printing. To find the keys (i.e. texts) in the
+        file:
+        "\n".join(FileImport().text_contents(FileImport().read_file(file)).keys()).
+
+        :param text_contents:
+        :param value:
+        :return:
+        """
+
+        return '\n'.join(text_contents[key])
+
+
+class CDLIImport(object):
+    """
+    The goal of this class is to take find the CDLI_number of a text and update
+    to its most recent iteration found in the CDLI github repository
+    (https://github.com/cdli-gh/data), or otherwise be able to pull any one
+    text from CDLI with the CDLI_number.
+    """
+    def __init__(self):
+        """
+        :param: empty.
+        """
+        self.import_text(cdli_number) = import_text(cdli_number)
```

Comment:
I'm not sure what's supposed to be happening here... your `__init__` method doesn't have any arguments so `cdli_number` isn't coming from anywhere... also there's no global method named `import_text`.  The `__init__` method should be for establishing some general parameters of the class object for instance identifying which CDLI data repository it's going to use for checking (i.e. just in case the CDLI moves its git repository) etc...

Reply:
adjusting with cdli_corpus comment below in mind